src/utils/helper.js

```diff
@@ -188,22 +188,32 @@ const uploadFile = ({
     })
 }
 
-const downloadWindowTitle = i18n.t('Loading exported data')
-const downloadWindowHtml = `
-<div style="height: 100%; width: 100%; display: flex; justify-content: center; align-items: center; color: rgb(33, 41, 52)">
-    <p>${downloadWindowTitle}</p>
-</div>
-`
-
 // call stub function if available
-const locationAssign = (url) => {
+const locationAssign = (relativeUrl) => {
     if (window.locationAssign) {
-        window.locationAssign(url)
+        window.locationAssign(relativeUrl)
     } else {
-        const downloadWindow = window.open(url, '_blank')
+        try {
+            const url = relativeUrl.startsWith('..')
+                ? new URL(relativeUrl, document.baseURI).href
+                : relativeUrl
+
+            const urlFilePart = new URL(url).pathname.split('/').pop()
+            const [, file, extension] = urlFilePart.match(/(^[^.]+)(\..+$)/)
 
-        downloadWindow.document.title = downloadWindowTitle
-        downloadWindow.document.body.innerHTML = downloadWindowHtml // does not work in Chrome
+            const downloadedFileName = `${file}${extension}`
```

Comment:
Is this needed? I thought the browser would handle the file-extension and we could just send the name, if we wanted?

Reply:
as in, I can only provide the `file` part and the browser will figure out the extension? I don't think so but I will give it a try

Reply:
you're right @Birkbjo - there was no need to add the extension, removed it now and addressed your comments